lib/config/presets/index.ts

```diff
@@ -338,24 +343,41 @@ export async function resolveConfigPresets(
           );
           throw error;
         }
+        if (shallowResolve && !shouldShallowResolve(preset)) {
```

Comment:
Shouldn't this happen before we fetch the preset? We discard it anyways, so we can save some calls and checks.

Reply:
I will create a separate small refactor for this for loop so it becomes more readable after these changes